src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -2053,7 +2053,7 @@ internal static bool CheckValEscape(SyntaxNode node, BoundExpression expr, uint
                     }
                     return true;
 
-                case BoundKind.StackAllocArrayCreation:
+                case BoundKind.ConvertedStackAllocExpression:
```

Comment:
I think we don't expect BoundStackAllocArrayCreation to reach here anymore. Consider adding case with Unreachable.
Same above.

Reply:
`BoundStackAllocArrayCreation` will stay in the tree in case of invalid code (that we cannot convert).

---
In reply to: [136683059](https://github.com/dotnet/roslyn/pull/21746#discussion_r136683059) [](ancestors = 136683059)